### PR TITLE
test: install auto-provisioning dependency

### PR DESCRIPTION
**Pull request by mhrabovcin**
**What type of PR is this?**

Chore

**What this PR does/ why we need it**:
Porting changes from the `release-1.2` branch that are necessary for running the latest `kommander` chart.

**Which issue(s) this PR fixes**:

Ported PR: https://github.com/mesosphere/kubeaddons-kommander/pull/193

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NO
```

**Checklist**

- [x] The commit message explains the changes and why are needed.
- [x] The code builds and passes lint/style checks locally.
- [ ] The relevant subset of integration tests pass locally.
- [ ] The core changes are covered by tests.
- [ ] The documentation is updated where needed.

### CODE REVIEW

File: test/kommander_test.go [hunk 1]
```diff
@@ -44,7 +72,48 @@ func TestKommanderGroup(t *testing.T) {
...
+		t.Fatal(err)
+	}
+
+	// `auto-provisioning` is now a dependency of the `kommander` and it is not installed
+	// as an addon but `konvoy` installs it using helm. Since the base cluster is
+	// a `kind` the `auto-provisioning` has to be installed manually. The
+	// `auto-provisioning` depends on `cert-manager` that must be installed before
+	// installing `auto-provisioning`. These jobs install `cert-manager` and wait
+	// for it to complete installation and then use `helm` to install `auto-provisioning`.
+	// Once these are in place other addons can be installed.
```
Comment (shaneutt) [inline test/kommander_test.go:106]: Appreciate the comments :+1:

File: test/kommander_test.go [hunk 2]
```diff
@@ -104,6 +173,7 @@ func TestKommanderGroup(t *testing.T) {
 	th := testharness.NewSimpleTestHarness(t)
 	th.Load(
 		addontesters.ValidateAddons(addons...),
+		installAutoProvisioning,
```
Comment (shaneutt) [inline test/kommander_test.go:176]: makes sense :+1:

**Review by shaneutt — APPROVED** [review]
Looks solid :+1: 

When CI is happy I'm happy.

**Review by hectorj2f — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+		git checkout $(KONVOY_SOURCE_VERSION) && \
+		git reset --hard origin/$(KONVOY_SOURCE_VERSION) && \
+		git clean -fdx
+endif
+
+BUILD_DIR ?= build
+
+auto-provisioning.prepare-chart: gitauth $(KONVOY_SOURCE)
+	mkdir -p $(BUILD_DIR)
+	$(MAKE) -C $(KONVOY_SOURCE) copy-charts BUILD_OUTPUT=$(PWD)/build
```
[the remark this answers is not in the document]
Comment (mhrabovcin) [reply Makefile:21]: This is just some `sed` work that takes less than a second.

**Pull request merged**